lemmy_apub/src/activities/receive/mod.rs

```diff
@@ -52,16 +55,12 @@ where
 {
   let cc = activity.cc().context(location_info!())?;
   let cc = cc.as_many().context(location_info!())?;
-  let community_followers_uri = cc
+  let community_uri = cc
     .first()
     .context(location_info!())?
     .as_xsd_any_uri()
     .context(location_info!())?;
-  // TODO: this is hacky but seems to be the only way to get the community ID
-  let community_uri = community_followers_uri
-    .to_string()
-    .replace("/followers", "");
-  let community = get_or_fetch_and_upsert_community(&Url::parse(&community_uri)?, context).await?;
+  let community = get_or_fetch_and_upsert_community(&community_uri, context).await?;
```

Comment:
This is the only logic change, as `cc` now contains the community ID, and not the inbox URL (so the replace isnt necessary anymore).